AWSAppSyncClient/Internal/AWSAppSyncClientInfo.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+import Foundation
+
+public protocol AWSAppSyncClientInfoProviding {
+    var apiUrl: String { get }
+    var region: String { get }
+    var authType: String { get }
+    var apiKey: String { get }
+}
+
+/**
+ * Configuration for AWSAppSyncClient as read from `awsconfiguration.json`
+ */
+public class AWSAppSyncClientInfo: AWSAppSyncClientInfoProviding {
+    private struct ConfigFileKeys {
+        static let root = "AppSync"
+        static let apiURL = "ApiUrl"
+        static let region = "Region"
+        static let authMode = "AuthMode"
+        static let apiKey = "ApiKey"
+    }
+
+    public let apiUrl: String
+    public let region: String
```

Comment:
Since this is a public class (which I mistakenly moved into the "Internal" directory and will restore), I don't really want to change the API without warning. However, I will rework the protocol to use strong types, provide a new instantiating type, and deprecate the `AWSAppSyncClientInfo` class (which doesn't really need to be a class anyway).